modules/backend/behaviors/FormController.php

```diff
@@ -339,7 +339,9 @@ public function update_onSave($recordId = null, $context = null)
      * `formAfterDelete`.
      *
      * @param int $recordId Record identifier
-     * @return mixed
+     * @return \Illuminate\Http\RedirectResponse|void
+     * @throws \October\Rain\Exception\ApplicationException
+     * @throws Exception
```

Comment:
@jacobdekeizer I'm leaning towards these being redundant on all methods in this class. They would make sense if we have multiple exception classes for individual pieces of the system, but we don't - we generally use `ApplicationException` and `SystemException` and that's it, so it can be a general assumption that a method *may* throw one of those exceptions at any given point.

Would you have any problems with removing them from your changes?

Reply:
Why should we make an assumption when we can typehint that this method throws these exceptions. Phpstorm will warn you when you don't catch an exception, if we remove this phpstorm will not warn you for uncaught exceptions. So I'm against removing them, because it helps the developer with what he can expect from these methods.

Reply:
@jacobdekeizer I recognise that PHPStorm can analyse this and give hints, but on the outset, if I'm reading that, those two lines in particular tell me absolutely nothing. It doesn't give me any indication as to *why* those exceptions might be thrown, especially when you view the code itself in which you find that the exception wouldn't be thrown by any of the method's code, and that PHPStorm found that the exception was instead potentially thrown by a method 3 or 4 levels down in the call stack.

It's also presumptuous in some cases, as there's potentially many more types of exceptions that can be thrown - especially when you bring plugins into the mix.

Reply:
That's why we have to typehint all exceptions, so we (as developers) aren't surprised with undocumented exceptions.

Reply:
@jacobdekeizer the point @bennothommo is trying to make is that at some point it's not possible to typehint all possible exceptions that could be thrown, especially when you start throwing event handling into the mix.

Reply:
@jacobdekeizer If you add context as to why those exceptions might be thrown (as per [@throws](https://docs.phpdoc.org/latest/references/phpdoc/tags/throws.html) documentation), it might make it more credible to keep these lines.